streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -123,14 +124,15 @@ void initializeNewTasks() {
                     log.debug("transitioning {} {} to restoring", taskTypeName, entry.getKey());
                     addToRestoring(entry.getValue());
                 } else {
-                    transitionToRunning(entry.getValue());
+                    readyPartitions.addAll(transitionToRunning(entry.getValue()));
```

Comment:
readyPartitions is already created.
It seems we don't need the extra Set created inside readyPartitions() - readyPartitions can be passed to readyPartitions()